www/java01/sp2/main.js

```diff
@@ -0,0 +1,386 @@
+
+  var pokemonWeight = document.createElement('p');
+  pokemonWeight.classList.add('pokemon-weight');
+  pokemonWeight.textContent = 'Weight: ' + pokemonData.weight;
+
+  statsContainer.appendChild(pokemonNameInStats);
+  statsContainer.appendChild(pokemonId);
+  statsContainer.appendChild(pokemonTypes);
+  statsContainer.appendChild(pokemonHeight);
+  statsContainer.appendChild(pokemonWeight);
+
+  statsAndImageContainer.appendChild(statsContainer);
+  statsAndImageContainer.appendChild(pokemonImage);
+
+  listItem.appendChild(statsAndImageContainer);
+
+  // Remove button
+  var removeButton = document.createElement('button');
+  removeButton.classList.add('pokemon-delete');
+  removeButton.textContent = 'Remove';
+  removeButton.addEventListener('click', function () {
+    console.log('Remove button clicked');
+    pokedexList.removeChild(listItem);
+
+    // Remove Pokemon from local storage
+    removeFromLocalStorage(pokemonData);
+
+    // Save Pokedex to local storage
+    savePokedexToLocalStorage();
+  });
+
+  listItem.appendChild(removeButton);
+
+  pokedexList.appendChild(listItem);
+}
+
+
+
+
+function savePokedexToLocalStorage() {
+  localStorage.setItem('pokedex', JSON.stringify(Array.from(pokedexList.children).map(getPokemonDataFromListItem)));
+}
+
+
+// Load Pokedex from local storage
+function loadPokedexFromLocalStorage() {
+  const savedPokedex = JSON.parse(localStorage.getItem('pokedex')) || [];
+  // Clear the existing content of the pokedexList before loading new items
+  pokedexList.innerHTML = '';
+
+  savedPokedex.forEach(function (pokemonData) {
+    addPokemonToPokedex(pokemonData);
+  });
+}
+
+
+
+// Remove Pokemon from local storage
+function removeFromLocalStorage(pokemon) {
+  const savedPokedex = JSON.parse(localStorage.getItem('pokedex')) || [];
+  const updatedPokedex = savedPokedex.filter((savedPokemon) => savedPokemon.id !== pokemon.id);
+  localStorage.setItem('pokedex', JSON.stringify(updatedPokedex));
+
+  // Save the updated Pokedex to local storage after removing the Pokemon
+  savePokedexToLocalStorage();
+}
+
+
+function getPokemonDataFromListItem(listItem) {
+  const pokemonData = {
+    id: listItem.getAttribute('data-id'),
+    name: (listItem.querySelector('.pokemon-name') || {}).textContent?.replace('Name: ', ''),
+    image: (listItem.querySelector('.pokemon-image') || {}).style.backgroundImage?.replace(/url\(["']?([^"']*)["']?\)/, '$1') || '',
+    types: (listItem.querySelector('.pokemon-types') || {}).textContent?.replace('Type(s): ', '').split(', ') || [],
+    height: (listItem.querySelector('.pokemon-height') || {}).textContent?.replace('Height: ', ''),
+    weight: (listItem.querySelector('.pokemon-weight') || {}).textContent?.replace('Weight: ', ''),
+  };
+  return pokemonData;
+}
+
+
+
+
+
+var addCard = function (pokemon) {
+  var card = document.createElement('div');
+  card.classList.add('card');
+  card.innerText = pokemon.name;

```

Comment:
nedavejte pokemon.name do innerText, vsichni to uvidi a hra bude moc easy

![image](https://github.com/nvbach91/4IZ268-2023-2024-ZS/assets/20724910/1d6657e9-1257-4995-93b7-6f626ddbd494)